docs/protocol/register_packages_on-chain.md

```diff
@@ -0,0 +1,95 @@
+This guide describes how to register software packages in the Autonolas Protocol using the [protocol frontend](https://protocol.autonolas.network/). You can register three types of packages: **services**, **agents** and **components**.
+
+## How packages are registered
+
+To register a package in the protocol, it must have been already published into a remote (IPFS) registry. Read [how to publish packages](https://docs.autonolas.network/open-autonomy/guides/publish_fetch_packages/) with the {{open_autonomy}} framework.
+
+The process of registering a package comprises the creation of a **metadata file** which contains a pointer to the IPFS hash of the published software package itself, plus some other additional information required by the protocol. This metadata file is also stored in the IPFS registry. The [protocol frontend](https://protocol.autonolas.network/) will automatically create and publish the metadata file based on the package/unit information and the information provided in the registration form. Then, it will interact with the protocol smart contracts to complete the registration of your package on-chain.
+
+Upon completing the registration process, you will have minted on-chain an NFT representing your package. See the figure below.
```

Comment:
We can drop on-chain here imo – it's implicit because you can't mint an NFT anywhere but on-chain.